kratos/utilities/data_type_traits.h

```diff
@@ -231,6 +252,27 @@ template<class TDataType, std::size_t TSize> class DataTypeTraits<array_1d<TData
     ///@name Public static operations
     ///@{
 
+    /**
+     * @brief Returns whther the given @ref TCheckIndex is dynamic.
+     *
+     * This method returns true if the @ref TCheckIndex of the dimension
+     * corresponds to a dynamic type. If not, this returns false.
+     *
+     * @tparam TCheckIndex          User input dimension index.
+     * @tparam TCurrentIndex        Used for the recursion only. Do not specify this.
+     * @return true                 If the input dimension index corresponds to dynamic data type.
+     * @return false                If the input dimension index corresponds to static data type.
+     */
+    template<unsigned int TCheckIndex, unsigned int TCurrentIndex = 0>
```

Comment:
if the user isn't supposed to touch this, don't expose it. You could hide this template param in an impl function. For example:
```cpp
template <unsigned TCheckIndex>
static constexpr bool IsDimensionDynamic()
{
    return DataTypeTraits::IsDimensionDynamicImpl<TCheckIndex,0>();
}

private:
template <unsigned TCheckIndex, unsigned TCurrentIndex>
static constexpr bool IsDimensionDynamicImpl()
{
    /**/
}
```

Reply:
Good point. and done :)